runtime/doc/diagnostic.txt

```diff
@@ -163,6 +163,37 @@ show a sign for the highest severity diagnostic on a given line: >lua
     }
 <
 
+                                          *diagnostic-handlers-example-loclist*
+To automatically update the |location-list| list with new diagnostics, we can
+create a new handler: >lua
+
+  vim.diagnostic.handlers.loclist = {
+    show = function(_, _, _, opts)
+      -- Generally don't want it to open on every update
+      opts.loclist.open = opts.loclist.open or false
+      vim.diagnostic.setloclist(opts.loclist)
+    end
+  }
+<
+
+The "hide" function can be omitted for this handler.
+
+Now, everytime the |location-list| is opened, it will show the most recent
+diagnostics. The handler accepts the same options as
+|vim.diagnostic.setloclist()| and can be configured using
+|vim.diagnostic.config()|: >lua
```

Comment:
This is thoughtfully worded, but still way too verbose. Examples should be as small as possible, and only essential things should be noted. Otherwise it just gets messy and hard to read and maintain.

```suggestion
                                          *diagnostic-loclist-example*
Whenever the |location-list| is opened, the following `show` handler will show
the most recent diagnostics: >lua

  vim.diagnostic.handlers.loclist = {
    show = function(_, _, _, opts)
      -- Generally don't want it to open on every update
      opts.loclist.open = opts.loclist.open or false
      vim.diagnostic.setloclist(opts.loclist)
    end
  }
<

The handler accepts the same options as |vim.diagnostic.setloclist()| and can be
configured using |vim.diagnostic.config()|: >lua
```